harness/tests/requirements/requirements-harness.txt

```diff
@@ -5,5 +5,6 @@ responses
 requests_mock
 coverage
 pytorch_lightning==1.5.9
-deepspeed==0.6.0
+deepspeed==0.8.3
+transformers>=4.8.2,<4.29.0
```

Comment:
this is pretty specific, would you mind including a comment explaining what is special about 4.29.0?  (and possibly 4.8.2, if it's a quick explanation)

Reply:
Not sure, I would need garrett to specify, this is going off of this comment in a PR a couple of days ago: https://github.com/determined-ai/determined/pull/6943#discussion_r1202950057 @garrett361

Reply:
I don't know this is necessary. In that comment I was just noting in that comment that I ran a similar example the other day and found that changes in a very recent (maybe `4.29.0`) release broke that example, and so there was some risk of that happening here, too.  

So, could remove the `<4.29.0` restriction if we verify that our examples still work with the latest `transformers`?

Reply:
Especially for pins that limit new versions, it's important to leave explanations of why they are in place, because otherwise it becomes [scary/dangerous for another developer to remove them](https://en.wikipedia.org/wiki/Wikipedia:Chesterton%27s_fence) months or years in the future.